app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/GoogleSignInWidget.kt

```diff
@@ -37,6 +39,14 @@ class GoogleSignInWidget : Fragment() {
             onLoginButtonPress()
         }
         updateUi()
+
+        val callback = object : OnBackPressedCallback(true){
+            override fun handleOnBackPressed() {
+                findNavController().popBackStack()
+            }
+        }
+
+        requireActivity().onBackPressedDispatcher.addCallback(callback)
```

Comment:
Is this necessary/well-located? the widget appears on the home screen, so shouldn't the home screen take care of that?

Reply:
true. I just put this in every fragment, but didn't consider that this widget wasn't part of the nav graph/ was handled by home screen